balance_service/tests.py

```diff
@@ -0,0 +1,661 @@
+                    "auth_url": "https://api.example.com:5000/v3",
+                    "region_name": "RegionOne",
+                },
+                "current_lease": {
+                    "start_date": lease_start,
+                    "end_date": lease_end,
+                    "project_id": "a0b86a98-b0d3-43cb-948e-00689182efd4",
+                    "user_id": "c631173e-dec0-4bb7-a0c3-f7711153c06c",
+                    "name": "my_lease",
+                    "reservations": [
+                        {
+                            "resource_type": "physical:host",
+                            "min": allocations_per_res,
+                            "max": allocations_per_res,
+                            "hypervisor_properties": "[]",
+                            "resource_properties": '["==", "$availability_zone", "az1"]',
+                            "id": str(j),
+                            "allocations": [
+                                {
+                                    "id": str(i),
+                                    "hypervisor_hostname": "32af5a7a-e7a3-4883-a643-828e3f63bf54",
+                                    "extra": {"availability_zone": "az1"},
+                                    "su_factor": su_factor,
+                                }
+                                for i in range(allocations_per_res)
+                            ],
+                        }
+                        for j in range(reservations)
+                    ],
+                },
+                "lease": {
+                    "start_date": update_start,
+                    "end_date": update_end,
+                    "project_id": "a0b86a98-b0d3-43cb-948e-00689182efd4",
+                    "user_id": "c631173e-dec0-4bb7-a0c3-f7711153c06c",
+                    "name": "my_lease",
+                    "reservations": [
+                        {
+                            "resource_type": "physical:host",
+                            "min": update_allocations_per_res,
+                            "max": update_allocations_per_res,
+                            "hypervisor_properties": "[]",
+                            "resource_properties": '["==", "$availability_zone", "az1"]',
+                            "id": str(j),
+                            "allocations": [
+                                {
+                                    "id": str(i),
+                                    "hypervisor_hostname": "32af5a7a-e7a3-4883-a643-828e3f63bf54",
+                                    "extra": {"availability_zone": "az1"},
+                                    "su_factor": update_su_factor,
+                                }
+                                for i in range(update_allocations_per_res)
+                            ],
+                        }
+                        for j in range(update_reservations)
+                    ],
+                },
+            }
```

Comment:
to comment, at a glance it's not clear to me which values are actually important, and which are just placeholders. Is this basically generating test data to help parametrize other tests?

Reply:
This is generating blazar API data to test the methods that our API view calls. https://docs.openstack.org/blazar/2024.1/admin/usage-enforcement.html

The most relevant parts are # of reservations, # of allocations in each reservation, and su factor, along with the lease duration.